status/controller.go

```diff
@@ -207,6 +209,28 @@ func (c *Controller[T]) Reconcile(ctx context.Context, req reconcile.Request) (r
 	return reconcile.Result{RequeueAfter: time.Second * 10}, nil
 }
 
+type GenericObjectController[T client.Object] struct {
+	*Controller[UnstructuredAdapter]
+}
+
+func NewGenericObjectController[T client.Object](client client.Client, eventRecorder record.EventRecorder) *GenericObjectController[T] {
+	return &GenericObjectController[T]{
+		Controller: NewController[UnstructuredAdapter](client, eventRecorder),
+	}
+}
+
+func (c *GenericObjectController[T]) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	return c.reconcile(ctx, req, lo.Must(NewUnstructuredAdapter(object.New[T]())))
+}
+
+func (c *GenericObjectController[T]) Register(_ context.Context, m manager.Manager) error {
```

Comment:
We can't drop this function unfortunately because it uses the `T` as the unstructured adapter if we don't write our own here -- which doesn't work for getting the name or generating the watch